velox/exec/TableScan.cpp

```diff
@@ -325,8 +333,7 @@ void TableScan::checkPreload() {
            this](const std::shared_ptr<connector::ConnectorSplit>& split) {
             preload(split);
 
-            executor->add([taskHolder = operatorCtx_->task(),
```

Comment:
Why don't we need `taskHolder = operatorCtx_->task()`?
Didn't we add this so that the task and its memory pool are not deleted before the preload?

Reply:
Never mind. The task is alive until all the preloads are canceled/completed. So this is fine.

Reply:
When the move() function is invoked before the prepare() function, the caller retrieves the result from move() without waiting for the completion of the prepare() function. So the task may finish before this lambda finishes, I think this is the reason taskHolder added before. 
`close` method would release all resources (task, etc.) from `AsyncSource` when task ending. So we don't need to add `taskHolder` here. 
Considering the case that `executor` here is busy or hang, tasks may finish but can not be deconstructed as they are referenced here. Associated memory would not be recycled unitl `executor` recovers. I think this should be considered as "memory leak" also. So I think this should be removed.